zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/clock/ClockProcessor.java

```diff
@@ -46,29 +47,56 @@ public ClockPinProcessor(
   public void processNewCommand(final TypedRecord<ClockRecord> command) {
     final long eventKey = keyGenerator.nextKey();
     final var clockRecord = command.getValue();
+    final var intent = (ClockIntent) command.getIntent();
 
-    applyClockModification(eventKey, clockRecord);
+    applyClockModification(eventKey, intent, clockRecord);
     if (command.hasRequestMetadata()) {
-      responseWriter.writeEventOnCommand(eventKey, ClockIntent.PINNED, clockRecord, command);
```

Comment:
Oh no, did I miss it? :see_no_evil:

Reply:
It wasn't caught by the test because we don't check the response, only the exported records, which were fine =/

I'll add support for response testing. Can't believe I missed that :facepalm: